File: 双指针 two pointers/0088. 合并两个有序数组/合并两个有序数组.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        if(!nums2.empty())
        {
            int j=0;
            for(int i=0;i<m+n;i++)
            {
                if(nums2[j]<nums1[i])
                {
                    nums1.insert(nums1.begin()+i,nums2[j]);
                    j++;
                    if(j==nums2.size())
                        break;
                }
            }
            while(j<nums2.size())
            {
                nums1.insert(nums1.begin()+m+j,nums2[j]);
                j++;
            }
            nums1.resize(m+n);
        }
        
    }
};

File: 双指针 two pointers/0088. 合并两个有序数组/合并两个有序数组.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        if(len(nums2)!=0):
            j=0
            i=0
            while i<m+n:
                if nums2[j]<nums1[i]:
                    nums1.insert(i,nums2[j])
                    j=j+1
                    if j==len(nums2):
                        break
                i=i+1
            while j<len(nums2):
                nums1.insert(m+j,nums2[j])
                j=j+1
            nums1[:]=nums1[0:m+n]